### PR TITLE
Fix some MSVC compilation failures

### PR DESCRIPTION
**Pull request by WilliamTambellini**
Fix some MSVC compilation failures

### CODE REVIEW

File: flashlight/memory/managers/DefaultMemoryManager.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #include <algorithm>
 #include <cstdlib>
+#include <iterator> // needed on MSVC to use back_insertor
```
Comment (jacobkahn) [inline flashlight/memory/managers/DefaultMemoryManager.cpp:17]: nit: we can remove the comment - we should have `<iterator>` regardless of the platform 👍

File: flashlight/memory/MemoryManagerInstaller.cpp [hunk 2]
```diff
@@ -47,7 +47,8 @@ MemoryManagerInstaller::MemoryManagerInstaller(
         "passed MemoryManagerAdapter is null");
   }
 
-  af_memory_manager interface = impl_->getHandle();
+  // Note: 'interface' is a define/keyword in MSVC, not to be used for var names
```
Comment (jacobkahn) [inline flashlight/memory/MemoryManagerInstaller.cpp:50]: nit: remove comment, see above

File: flashlight/memory/MemoryManagerAdapter.cpp [hunk 3]
```diff
@@ -14,12 +14,12 @@
 #include "flashlight/common/Utils.h"
 
 namespace fl {
-
+// note: 'interface' is a define/keyword for MSVC, not to be used for var names
```
Comment (jacobkahn) [inline flashlight/memory/MemoryManagerAdapter.cpp:17]: nit: remove comment - putting it in the commit description is fine :)

File: flashlight/memory/managers/DefaultMemoryManager.cpp (hunk 1 above)
Comment (WilliamTambellini) [reply flashlight/memory/managers/DefaultMemoryManager.cpp]: done

File: flashlight/memory/MemoryManagerInstaller.cpp (hunk 2 above)
Comment (WilliamTambellini) [reply flashlight/memory/MemoryManagerInstaller.cpp]: done

File: flashlight/memory/MemoryManagerAdapter.cpp (hunk 3 above)
Comment (WilliamTambellini) [reply flashlight/memory/MemoryManagerAdapter.cpp]: done

**Pull request closed without merging**